Makefile

```diff
@@ -200,8 +201,7 @@ destroy_%:
 		cd "infra/$*"; \
 		terraform workspace select $*-$$index-$$region; \
 		terraform destroy ${terraform_destroy_args} -var-file="../../terraform.tfvars.json" -var=$*_k8s_region=$$region -var=cluster_id=$$index; \
-		terraform workspace select default; \
-		terraform workspace delete ${terraform_workspace_args} $*-$$index-$$region; \
+		[ $$? -eq 0 ] && terraform workspace select default && terraform workspace delete ${terraform_workspace_args} $*-$$index-$$region; \
```

Comment:
Do not delete the workspace if the destroy operation failed. Since the entire block runs as a single command, `make` doesn't abort and continues anyway, so we need to explicitly short-circuit that to avoid losing the status when stuff is still there in the cloud provider.

This allows retrying the destroy operation if it failed, for example, because of an expired token.

Reply:
maybe we could pass instead the `-e` flag to the `sh` invocation to stop the execution on first error? with `@/bin/sh -e -c`

Reply:
This will attempt to destroy stuff from cloud providers in a loop, and we want it to be able to destroy as much as possible, so I'd rather not stop the loop but let it continue without messing the state for stuff that couldn't be deleted.

Reply:
thank you